Atomic.md

```diff
@@ -0,0 +1,278 @@
+Atomic placeholder operations in VFSForGit on MacOS
+===================================================
+
+A potential hole in our current design is in our creation of
+placeholders. Today, when creating a placeholder directory, at a high
```

Comment:
Minor comment (unrelated to the proposed fix):

> when creating a placeholder directory

I wasn't sure when I read this if "creating a placeholder directory" meant expanding a directory, or writing a placeholder for a directory.

The bullet points are not directory specific, and so you might want to simplify this to "when creating a placeholder"

Reply:
Correct, this is an issue about creating any placeholder. Not specific to placeholder directories.